### PR TITLE
Fix BlocksByRange RPC requests when count is 0

### PR DESCRIPTION
**Pull request by StefanBratanov**
## PR Description
We had an error processing RPC blocks by range request when count is set to 0
```
Unhandled error while processing request /eth2/beacon_chain/req/beacon_blocks_by_range/2/ssz_snappy","throwable":" java.lang.ArithmeticException: uint64 underflow
 tech.pegasys.teku.infrastructure.unsigned.UInt64.minus(UInt64.java:203)
tech.pegasys.teku.infrastructure.unsigned.UInt64.minus(UInt64.java:157)
tech.pegasys.teku.spec.datastructures.networking.libp2p.rpc.BeaconBlocksByRangeRequestMessage.getMaxSlot(BeaconBlocksByRangeRequestMessage.java:70)
```

We technically should be able to handle such requests even if they are deemed useless. We have requests limiting so we are not prone to DOS attacks. 

Had to also fix `BlobSidecarsByRange` since `maxSlot` was not calculated correctly for `count == 0`

Added tests for both blocks and blob sidecars to ensure they both work.

Did some other small refactors along the way.

## Fixed Issue(s)
N/A

## Documentation

- [X] I thought about documentation and added the `doc-change-required` label to this PR if updates are required.

## Changelog

- [X] I thought about adding a changelog entry, and added one if I deemed necessary.

### CODE REVIEW

File: networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/rpc/beaconchain/methods/BeaconBlocksByRangeMessageHandler.java [hunk 1]
```diff
@@ -183,7 +187,7 @@ private SafeFuture<RequestState> sendMatchingBlocks(
                       .map(MinimalBeaconBlockSummary::getSlot)
                       .orElse(ZERO);
               final NavigableMap<UInt64, Bytes32> hotRoots;
-              if (combinedChainDataClient.isFinalized(endSlot)) {
+              if (combinedChainDataClient.isFinalized(message.getMaxSlot())) {
```
Comment (gfukushima) [inline networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/rpc/beaconchain/methods/BeaconBlocksByRangeMessageHandler.java:185]: I'm not quite sure this will end up in the same slot number as the previous approach used in line 168. Was this change intended?

File: ethereum/spec/src/main/java/tech/pegasys/teku/spec/datastructures/networking/libp2p/rpc/BeaconBlocksByRangeRequestMessage.java [hunk 2]
```diff
@@ -67,7 +67,7 @@ public UInt64 getStep() {
   }
 
   public UInt64 getMaxSlot() {
-    return getStartSlot().plus(getCount().minus(1).times(getStep()));
+    return getStartSlot().plus(getCount().minusMinZero(1).times(getStep()));
```
Comment (gfukushima) [inline ethereum/spec/src/main/java/tech/pegasys/teku/spec/datastructures/networking/libp2p/rpc/BeaconBlocksByRangeRequestMessage.java:70]: Alright, so after treading the spec I think this won't actually have any unexpected effects because step is always 1 so it should be fine. Maybe worth the change in the order so we don't trip into a issue if `step` for some reason gets re-used for another purpose.

Comment (StefanBratanov) [reply ethereum/spec/src/main/java/tech/pegasys/teku/spec/datastructures/networking/libp2p/rpc/BeaconBlocksByRangeRequestMessage.java:70]: Actually `final UInt64 endSlot = startSlot.plus(step.times(count)).minus(ONE);` won't be correct when step > 1, thus I replaced it with `message.getMaxSlot()` which is more accurate.

Comment (StefanBratanov) [reply ethereum/spec/src/main/java/tech/pegasys/teku/spec/datastructures/networking/libp2p/rpc/BeaconBlocksByRangeRequestMessage.java:70]: There is a test in `BeaconBlocksByRangeRequestMessageTest` which covers this method

**Comment by StefanBratanov** [comment]
@gfukushima after some discussion with Enrico and Mehdi, we decided to change the logic to be able to handle such requests, just letting you know

File: ethereum/spec/src/main/java/tech/pegasys/teku/spec/datastructures/networking/libp2p/rpc/BeaconBlocksByRangeRequestMessage.java (hunk 2 above)
Comment (gfukushima) [reply ethereum/spec/src/main/java/tech/pegasys/teku/spec/datastructures/networking/libp2p/rpc/BeaconBlocksByRangeRequestMessage.java:70]: Should we use the same logic for `BlobSidecarsByRangeRequestMessage.getMaxSlot()` then?

Comment (StefanBratanov) [reply ethereum/spec/src/main/java/tech/pegasys/teku/spec/datastructures/networking/libp2p/rpc/BeaconBlocksByRangeRequestMessage.java:70]: There is no `step` there, so we are safe

File: networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/rpc/beaconchain/methods/BeaconBlocksByRangeMessageHandler.java (hunk 1 above)
Comment (mehdi-aouadi) [reply networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/rpc/beaconchain/methods/BeaconBlocksByRangeMessageHandler.java:185]: The `getMaxSlot` gives a different result but is correct. IMU the `endSlot` calculation wasn't correct and returned `maxSlot + 1`. The first blocks (from the requested count) is always at the `startSlot` so we should first subtract 1 from the `count` to calculate the `endSlot`/`maxSlot`

File: storage/src/main/java/tech/pegasys/teku/storage/server/CombinedStorageChannelSplitter.java [hunk 3]
```diff
@@ -235,7 +235,7 @@ public SafeFuture<List<SlotAndBlockRootAndBlobIndex>> getAllBlobSidecarKeys(fina
 
   @Override
   public SafeFuture<List<SlotAndBlockRootAndBlobIndex>> getBlobSidecarKeys(
-      UInt64 startSlot, UInt64 endSlot, UInt64 limit) {
+      UInt64 startSlot, UInt64 endSlot, long limit) {
```
Comment (mehdi-aouadi) [inline storage/src/main/java/tech/pegasys/teku/storage/server/CombinedStorageChannelSplitter.java]: Could add the missing finals here

**Review by mehdi-aouadi — APPROVED** [review]
LGTM
